### PR TITLE
v3.0 java

### PR DESCRIPTION
**Pull request by ksyeo1010**

### CODE REVIEW

File: binding/java/src/ai/picovoice/porcupine/Porcupine.java [hunk 1]
```diff
@@ -31,6 +31,8 @@ public class Porcupine {
     public static final String MODEL_PATH;
     public static final HashMap<BuiltInKeyword, String> BUILT_IN_KEYWORD_PATHS;
 
+    private static String sdk = "java";
```
Comment (laves) [inline binding/java/src/ai/picovoice/porcupine/Porcupine.java:34]: update the copyright to 2023

**Review by laves — APPROVED** [review]

File: binding/java/src/ai/picovoice/porcupine/exception/PorcupineException.java [hunk 2]
```diff
@@ -11,11 +11,44 @@
...
+        if (messageStack != null) {
+            if (messageStack.length > 0) {
+                sb.append(":");
+                for (int i = 0; i < messageStack.length; i++) {
+                    sb.append(String.format("\n  [%d] %s", i, messageStack[i]));
+                }
+            } else {
+                sb.append(".");
+            }
+        }
```
Comment (laves) [inline binding/java/src/ai/picovoice/porcupine/exception/PorcupineException.java:51]: I don't think placing the period is necessary
```suggestion
        if (messageStack != null) {
            if (messageStack.length > 0) {
                sb.append(":");
                for (int i = 0; i < messageStack.length; i++) {
                    sb.append(String.format("\n  [%d] %s", i, messageStack[i]));
                }
            }
        }
```

Comment (ksyeo1010) [reply binding/java/src/ai/picovoice/porcupine/exception/PorcupineException.java:51]: I added period everywhere, those the prs I merged have the period?

Comment (ksyeo1010) [reply binding/java/src/ai/picovoice/porcupine/exception/PorcupineException.java]: oh ok, weird I put it on java only haha

**Pull request merged**